### PR TITLE
ROC-5470: Failed/Successful mediation events

### PR DESCRIPTION
**Pull request by SanchitaMoj**
**Confirm that:**

- [ ] *VERSION* has been bumped (only for definition file changes)
- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)

### Change description ###

**Does this PR introduce a breaking change?**

```
- [ ] Yes
- [ ] No
```

### CODE REVIEW

File: definition/data/sheets/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -106,6 +106,12 @@
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "ReferredToMediation", "UserRole": "caseworker-cmc-judge", "CRUD": "R"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "UserRole": "caseworker-cmc-judge", "CRUD": "R"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "UserRole": "caseworker-cmc", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "UserRole": "caseworker-cmc-legaladvisor", "CRUD": "R"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "UserRole": "citizen", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "FailedMediation", "UserRole": "caseworker-cmc-judge", "CRUD": "R"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "FailedMediation", "UserRole": "caseworker-cmc", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "FailedMediation", "UserRole": "caseworker-cmc-legaladvisor", "CRUD": "R"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "FailedMediation", "UserRole": "citizen", "CRUD": "CR"},
```
Comment (karthikkannankainos) [inline definition/data/sheets/AuthorisationCaseEvent.json]: Just Read for Citizen pls.

File: definition/data/sheets/AuthorisationCaseEvent.json [hunk 2]
```diff
@@ -106,6 +106,12 @@
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "ReferredToMediation", "UserRole": "caseworker-cmc-judge", "CRUD": "R"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "UserRole": "caseworker-cmc-judge", "CRUD": "R"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "UserRole": "caseworker-cmc", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "UserRole": "caseworker-cmc-legaladvisor", "CRUD": "R"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SuccessfulMediation", "UserRole": "citizen", "CRUD": "CR"},
```
Comment (karthikkannankainos) [inline definition/data/sheets/AuthorisationCaseEvent.json]: just Read for Citizen pls.

File: definition/data/sheets/CaseEventToComplexTypes.json [hunk 3]
```diff
@@ -5,5 +5,14 @@
...
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "respondents","ListElementCode": "partyName","EventElementLabel": "Defendant name","FieldDisplayOrder": "","DisplayContext": "READONLY"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "respondents","ListElementCode": "mediationSettlementReachedAt","EventElementLabel": "Mediation settlement agreed on","FieldDisplayOrder": 1,"DisplayContext": "MANDATORY"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "staffUploadedDocuments","ListElementCode": "documentName","EventElementLabel": "Document name","FieldDisplayOrder": 1,"DisplayContext": "OPTIONAL"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "staffUploadedDocuments","ListElementCode": "documentType","EventElementLabel": "Document type","FieldDisplayOrder": 1,"DisplayContext": "OPTIONAL"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "staffUploadedDocuments","ListElementCode": "documentLink","EventElementLabel": "Mediation agreement link","FieldDisplayOrder": 1,"DisplayContext": "OPTIONAL"},
```
Comment (karthikkannankainos) [inline definition/data/sheets/CaseEventToComplexTypes.json]: documentLink refers to DM link?

Comment (SanchitaMoj) [reply definition/data/sheets/CaseEventToComplexTypes.json]: when you upload via CCD, the document is first pushed to doc store and then the returned link is displayed, so yes

File: definition/data/sheets/CaseEventToComplexTypes.json [hunk 4]
```diff
@@ -5,5 +5,14 @@
...
-]
\ No newline at end of file
+    {"ID": "ScannedDocument","CaseEventID": "ReviewedPaperResponse","CaseFieldID": "scannedDocuments","ListElementCode": "subtype","EventElementLabel": "Document Subtype","EventHintText": "","FieldDisplayOrder": 3,"DisplayContext": "MANDATORY"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "respondents","ListElementCode": "partyName","EventElementLabel": "Defendant name","FieldDisplayOrder": "","DisplayContext": "READONLY"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "respondents","ListElementCode": "mediationSettlementReachedAt","EventElementLabel": "Mediation settlement agreed on","FieldDisplayOrder": 1,"DisplayContext": "MANDATORY"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "staffUploadedDocuments","ListElementCode": "documentName","EventElementLabel": "Document name","FieldDisplayOrder": 1,"DisplayContext": "OPTIONAL"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "staffUploadedDocuments","ListElementCode": "documentType","EventElementLabel": "Document type","FieldDisplayOrder": 1,"DisplayContext": "OPTIONAL"},
```
Comment (karthikkannankainos) [inline definition/data/sheets/CaseEventToComplexTypes.json]: do we need to add the documentType fixed List?

File: definition/data/sheets/CaseEventToComplexTypes.json [hunk 5]
```diff
@@ -5,5 +5,14 @@
...
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "staffUploadedDocuments","ListElementCode": "documentLink","EventElementLabel": "Mediation agreement link","FieldDisplayOrder": 1,"DisplayContext": "OPTIONAL"},
+    {"ID": "Respondant","CaseEventID": "SuccessfulMediation","CaseFieldID": "staffUploadedDocuments","ListElementCode": "receivedDatetime","EventElementLabel": "Agreement received on","FieldDisplayOrder": 1,"DisplayContext": "OPTIONAL"},
+    {"ID": "Respondant","CaseEventID": "FailedMediation","CaseFieldID": "respondents","ListElementCode": "partyName","EventElementLabel": "Defendant name","FieldDisplayOrder": 2,"DisplayContext": "READONLY"},
+    {"ID": "Respondant","CaseEventID": "FailedMediation","CaseFieldID": "respondents","ListElementCode": "mediationFailedReason","EventElementLabel": "Mediation failed reason","FieldDisplayOrder": 3,"DisplayContext": "MANDATORY"},
```
Comment (karthikkannankainos) [inline definition/data/sheets/CaseEventToComplexTypes.json]: ',' comma at last element. this is failing the build

File: definition/data/sheets/AuthorisationCaseField.json [hunk 6]
```diff
@@ -204,6 +204,7 @@
   {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "evidenceHandled", "UserRole": "caseworker-cmc", "CRUD": "CRUD"},
   {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "paperResponseType", "UserRole": "caseworker-cmc", "CRUD": "CRUD"},
   {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "paperResponseType", "UserRole": "citizen", "CRUD": "CRUD"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "state", "UserRole": "caseworker-cmc", "CRUD": "CU"},
```
Comment (dharmendrak) [inline definition/data/sheets/AuthorisationCaseField.json]: why we are adding state In case

Comment (karthikkannankainos) [reply definition/data/sheets/AuthorisationCaseField.json]: @dharmendrak there is an issue with CCD that for the About To Submit call back if we are going to change the state, we have to pass it as the case detail. So to overcome we have to have this field. I explained to @ckalista and he had taken it to CCD. They confirmed that its existing ticket.

**Pull request merged**